### ISSUE TITLE
Update the docs code blocks to provide both Stitches and CSS examples.

### ISSUE THREAD

**Issue by colmtuite** [radix-ui/primitives#1728]
### Why?

To make adoption as easy as possible, we must maximise copy/pasteability. While Stitches is a popular lib, the vast majority of devs still use either plain CSS, Tailwind, or other CSS-in-JS libs. Furthermore, many devs have either never used Stitches, or never even heard of it.

So, when new devs land on the docs site, they are commonly confused by the Stitches code examples. They're often not sure whether these Stitches code snippets are part of Radix, or separate. They don't know that Radix is compatible with any styling solution. So it causes confusion right from the beginning.

By changing the docs code examples to use plain CSS, and provide Stitches an option, we're catering to many more people.

### Solution

Add a language toggle to hero code examples so users can switch between different styling libs like Stitches, CSS, Tailwind etc.

### Language Picker UI

The UI should be similar to the UI for our new Docs. So, a Select menu on the right for choosing the language/lib, and tabs on the left for different files.

Note: Some languages will require just a single file, like Stitches, because both styles and markup are collocated. Other languages will require two files, like CSS, where styles and markup are separated. So the language selector menu should control the tabs UI.

<img width="762" alt="Screenshot 2022-10-18 at 19 33 17" src="https://user-images.githubusercontent.com/805073/196515427-8fbadcd2-f9ab-4b08-9a67-d68b683bda0b.png">

### Notes

- We need to figure out a solution for storing the multiple languages.
- Rewriting the examples in plain CSS should be straightforward. The only variables/tokens used in the code examples are colors. All other properties which accept custom values have their values hard-coded. The colors use imported Radix Colors. So it should be a case of just using the CSS import for Radix Colors instead, and replacing all of the styles with CSS instead of Stitches.
- Language switcher controls code block / tabs UI. Some languages need tabs, others don't.
- It's worth noting that there will be a long-term maintenance burden with this work. When we update a hero code example, or introduce a new component, any changes/updates will need to be made across all languages we support. However, I don't anticipate changes like these often. Just something to be aware of.

### Tailwind

Introducing Tailwind will be a lot trickier.

Tailwind pain points:

- Currently, our examples use a 5px space scale, to accommodate our 15px icons. Tailwind does not support such a space scale. So we will either need to redesign all elements on a 4px space scale, then resize all Radix icons from 15px to 16px, or we'll need to customise the Tailwind config file to use a 5px space scale. If we customise the TW space scale, we lose a lot of copy/pasteability, because close to zero TW users will be using our 5px scale.
- The same goes for TW colors. They use a 10 step scale, Radix Colors uses a 12 step scale. So we will either need to redesign all of our demos to use 10 steps, or extend the TW color scales with two additional steps...again breaking copy/pasteability.

### Approach / Timeline and Tailwind

With the above in mind, I propose we go ahead with the Stitches, CSS, and UI work now, and release it. Then when we're finished, we can take a deeper look at Tailwind.

### Who does this impact? Who is this for?

All docs users.

**Comment by benoitgrelard** [comment]
I didn't realise you meant we should also keep stitches examples, I thought we were replacing with CSS.

Is the idea that we would show the multiple languages only for the top-level hero demos?
What do we do about all the other codeblocks? CSS only?

**Comment by colmtuite** [comment]
Yes CSS only, chatted with Andy briefly about this yesterday. The other examples/blocks are mostly just React/JSX, so I think it's fine to just use CSS for the few code blocks that require styles.

**Comment by benoitgrelard** [comment]
Yeah that makes sense.
What about when there is some CSS involved? I guess this will needs tabs?

https://www.radix-ui.com/docs/primitives/components/dropdown-menu#collision-aware-animations

**Comment by colmtuite** [comment]
Yep, tabs UI for these ones. But CSS only is perfectly fine imo.
